gtest/src/check.rs

```diff
@@ -123,15 +125,58 @@ fn check_messages(
             messages.len(),
         ))
     } else {
+        let mut expected_messages: Vec<sample::Message> = expected_messages.into();
+        let mut messages: Vec<Message> = messages.into();
+
         expected_messages
-            .iter()
-            .zip(messages.iter())
+            .iter_mut()
+            .zip(messages.iter_mut())
             .enumerate()
             .for_each(|(position, (exp, msg))| {
+                let meta_type = if exp.init.unwrap_or(false) {
+                    MetaType::InitOutput
+                } else {
+                    MetaType::Output
+                };
+
                 if exp
                     .payload
-                    .as_ref()
-                    .map(|payload| !payload.equals(msg.payload.as_ref()))
+                    .as_mut()
+                    .map(|payload| match payload {
+                        PayloadVariant::Custom(_) => {
+                            progs_n_paths
+                                .iter()
+                                .find(|v| ProgramId::from(v.1) == msg.source())
+                                .map(|v| {
+                                    let path: String = v.0.replace(".wasm", ".meta.wasm");
+
+                                    let json = MetaData::Json(
+                                        String::from_utf8(payload.to_bytes()).unwrap(),
+                                    );
+
+                                    let bytes = json
+                                        .convert(&path, meta_type.clone())
```

Comment:
We could either derive `Copy` (it is ok for primitive types) for `MetaType` or just pass a reference to avoid cloning.
```suggestion
                                        .convert(&path, &meta_type)
```

Reply:
I would prefer use references